pkg/templates/nstemplatetiers/nstemplatetier_generator_test.go

```diff
@@ -140,6 +175,27 @@ func TestCreateOrUpdateResources(t *testing.T) {
 
 			// then
 			require.NoError(t, err)
+			// verify that all TemplateTier CRs were updated
+			tierTmpls = toolchainv1alpha1.TierTemplateList{}
+			err = clt.List(context.TODO(), &tierTmpls, client.InNamespace(namespace))
+			require.NoError(t, err)
+			require.Len(t, tierTmpls.Items, len(testnstemplatetiers.AssetNames())-1) // exclude `metadata.yml` from the AssetNames, it does not result in a TemplateTier resource
+			for _, tierTmpl := range tierTmpls.Items {
+				assert.Equal(t, int64(2), tierTmpl.ObjectMeta.Generation)
+			}
+
+			for _, tierName := range []string{"advanced", "basic", "team", "nocluster"} {
+				tier := toolchainv1alpha1.NSTemplateTier{}
+				err = clt.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: tierName}, &tier)
+				require.NoError(t, err)
+				// verify that the generation was increased
+				assert.Equal(t, int64(2), tier.ObjectMeta.Generation)
+				// verify their new namespace revisions
+				for _, ns := range tier.Spec.Namespaces {
+					assert.Equal(t, nstemplatetiers.ExpectedRevisions[tierName][ns.Type], ns.Revision)
+					assert.NotEmpty(t, nstemplatetiers.ExpectedRevisions[tierName][ns.Type], ns.Template)
+				}
+			}
```

Comment:
This looks like a duplication of the following lines 200-211

Reply:
removed